ollama/_utils.py

```diff
@@ -0,0 +1,81 @@
+from __future__ import annotations
+from collections import defaultdict
+import inspect
+from typing import Callable, Union
+
+import pydantic
+from ollama._types import Tool
+
+
+def _parse_docstring(doc_string: Union[str, None]) -> dict[str, str]:
+  parsed_docstring = defaultdict(str)
+  if not doc_string:
+    return parsed_docstring
+
+  lowered_doc_string = doc_string.lower()
```

Comment:
What's the objective with lowering the entire docstring? AFAICT you only need specific values like `Args:` and `Returns:` etc. lowered

Reply:
Initially thought that if we were calling `.lower()` on the same thing multiple times it would be better do just lower it once. Don't think I ended actually needing it on the whole docstring so will probably change it.